src/Framework/Profiler/EvaluationLocation.cs

```diff
@@ -163,31 +163,33 @@ public override bool Equals(object obj)
                     EvaluationDescription == other.EvaluationDescription &&
                     string.Equals(File, other.File, StringComparison.OrdinalIgnoreCase) &&
                     Line == other.Line &&
-                    ElementName == other.ElementName;
+                    ElementName == other.ElementName &&
+                    ElementOrCondition == other.ElementOrCondition &&
+                    IsElement == other.IsElement;
             }
             return false;
         }
 
         /// <nodoc/>
-        public override int GetHashCode()
+        public override string ToString()
         {
-            var hashCode = 590978104;
-            hashCode = hashCode * -1521134295 + base.GetHashCode();
-
-            hashCode = hashCode * -1521134295 + EqualityComparer<EvaluationPass>.Default.GetHashCode(EvaluationPass);
-            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(EvaluationDescription);
-            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(File?.ToLower());
-            hashCode = hashCode * -1521134295 + EqualityComparer<int?>.Default.GetHashCode(Line);
-            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(ElementName);
-            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(ElementOrCondition);
-
-            return hashCode;
+            return $"{EvaluationDescription ?? string.Empty}\t{File ?? string.Empty}\t{Line?.ToString() ?? string.Empty}\t{ElementName ?? string.Empty}\tIsElement?{IsElement}\tElementOrCondition:{ElementOrCondition}\t{this.EvaluationDescription}";
         }
 
         /// <nodoc/>
-        public override string ToString()
+        public override int GetHashCode()
         {
-            return $"{EvaluationDescription ?? string.Empty}\t{File ?? string.Empty}\t{Line?.ToString() ?? string.Empty}\t{ElementName ?? string.Empty}";
+            unchecked
+            {
+                var hashCode = (int) EvaluationPass;
+                hashCode = (hashCode * 397) ^ (EvaluationDescription != null ? EvaluationDescription.GetHashCode() : 0);
+                hashCode = (hashCode * 397) ^ (File != null ? File.GetHashCode() : 0);
+                hashCode = (hashCode * 397) ^ Line.GetHashCode();
+                hashCode = (hashCode * 397) ^ (ElementName != null ? ElementName.GetHashCode() : 0);
```

Comment:
>(ElementName != null ? ElementName.GetHashCode() [](start = 46, length = 48)

just curious, why the change? The default equality comparer does exactly that (the null check).

And I think the original code was what resharper generates by default

Reply:
I suspected this could be the cause of the data loss I was seeing. Both the GetHashCode and equality were not completely correct (GetHashCode was missing members). I regenerated with resharper now it's more like it was before. Honestly don't have any preference, I just 1) don't want to write it and 2) I want it to take all the fields into account :). Especially in this case where it's used as a Dictionary key.